### ISSUE TITLE
Show Parent Functions' Cross-References in Rule Classification

### ISSUE THREAD

**Issue by haeter525** [quark-engine/quark-engine#191]
**Is your feature request related to a problem? Please describe.**

The current rule classification doesn't consider references between parent functions. If a parent function calls another to perform a behavior, it is difficult to check this behavior in the rule classification report.

**Describe the solution you’d like**

I propose a new feature that lists cross-references between parent functions in the rule classification report.
The feature adds extra descriptions to the Crime Description field to show what functions are called by the parent function.

Here is the pattern.

```
* Call {A function called by the parent function}
```

---

Imagine a scenario.

+ `Parent Function 2` does `Behavior 2`.
+ `Parent Function 1` does `Behavior 1` and calls `Parent Function 2`.

For `Parent Function 1`, the current rule classification table below lists one description in the Crime Description field.

```
+-------------------+------------------------------+
| Parent Function   | Parent Function 1            |
+-------------------+------------------------------+
| Crime Description | * Description for Behavior 1 |
+-------------------+------------------------------+
```

The feature adds a line (Line 5) to indicate its reference to `Parent Function 2`.

```
+-------------------+------------------------------+
| Parent Function   | Parent Function 1            |
+-------------------+------------------------------+
| Crime Description | * Description for Behavior 1 |
|                   | * Call Parent Function 2     | <--Add_by_this_feature.
+-------------------+------------------------------+
```

The feature also applies to the JSON version of the rule classification report.

```json
{
    "parent": "Parent Function 1",
    "crime": [
        "Behavior 1",
        "Call Parent Function 2"       # <--Add_by_this_feature.
    ]
}
```

**How can one use this feature?**

According to the additional descriptions, users can refer to the report of the called parent function. Take `Parent Function 1` as an example. Here is the report of  `Parent Function 2`.

```
+-------------------+----------------------------+
| Parent Function   | Parent Function 2          |
+-------------------+----------------------------+
| Crime Description | * Behavior 2 Description   |
+-------------------+----------------------------+
```
Now, the user knows that Parent Function 1 does two behaviors instead of one. With this feature, the user can see how the parent functions interact with each other.

**Will all referenced functions be listed in the field?**

No, only those functions that do at least one behavior are visible.

**Comment by haeter525** [comment]
Hi @krnick ,

Can you kindly take a look at this feature? If you think it is OK, please assign it to me.

Thank you!

**Comment by krnick** [comment]
Closed with #192
